group-aws/utility-modules.md

```diff
@@ -0,0 +1,104 @@
+# Utility modules.
+
+
+## Introduction
+
+The AWS utility modules provide functions which are designed to be reused between different AWS
+modules but not other Ansible modules (in which case the function could be shared in a different
+module).
+
+## Criteria for utility functions and classes
+
+* The functions will be copied with every AWS module which includes the utility module so must be kept lightweight.
+* Changes made for one module may affect another module which the developer is not using and may not even have the ability to test
+* Different modules may use the same utility function in different ways making the development of the functions more subtle and difficult. 
+
+For this reason the utility modules are expected to have a smaller amount of higher quality code. 
+
+Criteria for putting functions into the module_utils/aws directory 
+
+* Used in at least two modules
+* Reasonably clear, sensible and agreed upon interface
+* Don't know of a reason to think it will become obsolete soon
+* 90% or more unit test coverage (line)
+
+## Planned Layout
+
+The layout is subject to change, probably until just before the 2.5 release (at which point there
+should be a stable interface document).  Modules included in Ansible will get refactored together
+with the change.  Pull requests will have to be refactored before being accepted.  Modules ourside
+the main repository which are already using this interface will have to be refactored by their
+authors.
+
+    aws/
+        core.py - Basic AWS modules 
+            class AnsibleAWSModule - class to use for handling module set up'; based off AnsibleModule
+	    	.params() - returns the parameter of the module
```

Comment:
I'm thinking this should be an attribute method, so you don't have to distinguish module.params and awsmodule.params() usage. I can see that leading to some fatfingering.